cvat-ui/src/components/import-dataset/import-dataset-modal.tsx

```diff
@@ -66,67 +277,71 @@ function ImportDatasetModal(props: StateToProps): JSX.Element {
     } = props;
     const [form] = Form.useForm();
     const dispatch = useDispatch();
-    // TODO useState -> useReducer
-    const [instanceType, setInstanceType] = useState('');
-    const [file, setFile] = useState<File | null>(null);
-    const [selectedLoader, setSelectedLoader] = useState<any>(null);
-    const [useDefaultSettings, setUseDefaultSettings] = useState(true);
-    const [defaultStorageLocation, setDefaultStorageLocation] = useState(StorageLocation.LOCAL);
-    const [defaultStorageCloudId, setDefaultStorageCloudId] = useState<number | undefined>(undefined);
-    const [helpMessage, setHelpMessage] = useState('');
-    const [selectedSourceStorageLocation, setSelectedSourceStorageLocation] = useState(StorageLocation.LOCAL);
-    const [uploadParams, setUploadParams] = useState<UploadParams>({
-        convMaskToPoly: true,
+
+    const [state, dispatchReducer] = useReducer(reducer, {
```

Comment:
Lets use a bit different notation, `dispatch` as local dispatch, and `appDispatch` as main app dispatch